packages/dds/tree/src/shared-tree-core/branch.ts

```diff
@@ -492,6 +473,41 @@ export class SharedTreeBranch<TEditor extends ChangeFamilyEditor, TChange> {
 		return rebaseResult;
 	}
 
+	/**
+	 * Replace a range of commits on this branch with a single commit composed of equivalent changes.
+	 * @param startCommit - All commits after (but not including) this commit, up to and including the branch's head, will be squashed.
```

Comment:
`startCommit` seems like a weird name for something that's not included in the range of commits to be squashed.
Ideally we'd pass in the first commit that gets squashed, but I understand that's not practical.
Maybe we can rename the method `squashAfter` or `squashUpTo`.

Another approach may be to pass in the number of commits you want squashed. Would that be practical?
It's probably best not to worry too much since you're just trying to make progress on the refactor here.

Reply:
I like `squashAfter`. Eventually, it might make sense to pass in a count - but I am not sure until everything is settled. My concern would be that code outside of `squash` needs to iterate the commits to compute the count, and then the code inside of count needs to iterate the commits again according to the count. In that case, it would have been better to pass in the target commit (the current API) and only iterate once. I'm not sure but I'll keep it in mind in case a future change makes sense.